parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -341,8 +397,16 @@ private void readObjectNoData() throws ObjectStreamException {
 
   }
 
-  public static Binary fromByteBuffer(final ByteBuffer value) {
-    return new ByteBufferBackedBinary(value);
+  private static Binary fromByteBuffer(final ByteBuffer value, final boolean isUnmodifiable) {
+    return new ByteBufferBackedBinary(value, isUnmodifiable);
+  }
+
+  public static Binary fromReusedByteBuffer(final ByteBuffer value) {
+    return fromByteBuffer(value, false);
+  }
+
+  public static Binary fromUnmodifiedByteBuffer(final ByteBuffer value) {
```

Comment:
I don't think `fromUnmodifiedByteBuffer` is clear enough. Unmodified makes it sound like it is not modified when it is created, but might be afterward. I think const or constant gets closer, but what we want is to signal that this will not be reused or rewritten. Immutable is too strong because it has a formal definition in Java.

What about using `fromReusableByteBuffer` and `fromConstantByteBuffer`?